### ISSUE TITLE
Pattern to manage "external" resources

### ISSUE THREAD

**Issue by pmuens** [serverless/components#437]
We should define a common pattern / way to manage "external" resources which were not deployed via the Componens project.

**Comment by laardee** [comment]
What I was thinking was that should there be an `import` or `sync` feature which would fetch the current configuration from deployed resources and populate the serverless.yml with those? That way the manually created or altered resources would have up-to-date config in serverless.yml.


Then another thing. Should the components also remove the "imported" resource, e.g. if multiple serverless.yml has the same resource included. For example I could have a dynamo table imported to multiple services so that I get the arns and other things easily outputted from that. Then if I need to remove the other service, it shouldn't delete the table, because it is still in use. Should there be a `deletionPolicy` type of definition in the component. It could be e.g. `retain` like in cloudformation, or `imported/retain` if more granual configuration is needed.

**Comment by eahefnawy** [comment]
@laardee technically we have this `sync` functionality in most components (depends on the use case). Here's an example:
https://github.com/serverless-components/aws-lambda/blob/master/serverless.js#L107

We don't auto populate serverless.yml per se (because we also support programatic usage), but it merges with the inputs to reflect the actual state of the resource in the provider

**Comment by laardee** [comment]
The sync would be usefull in the cases if you are planning to delete the stack and redeploy it, and you have imported resources defined. Otherwise you have only partial definitions on the serverless.yml. That is if the remove is going to delete the imported resources also.

**Comment by laardee** [comment]
@eahefnawy Now I know what I was thinking differently from the logic that e.g. Lambda component has currently. I was at the conference yesterday when I wrote previous comments, so I was a bit distracted, sorry about that 😬 

The Lambda component currently uses only existing arn, and some other params, but does not merge other configuration from the existing lambda and overwrites those on deploy with defaults. 

What I was thinking about how it could work, is that when you hook existing manually created resources to components, the current configuration would not be overwritten with defaults but merged with existing and only updated with new. In that case, if you have only partial config in the serverless.yml the components cannot redeploy the resource with the same config if it is removed and the sync would be somewhat useful.

As a use case, there could be a CloudFront distribution that I've set up manually, with all the behaviours, origins, error responses, and other settings defined. Then I want to deploy Lambda@Edge function and attach it to the distribution with components, and I need to recreate the whole configuration to serverless.yml. Which I would rather let the components do, either with sync/import feature or if the deploy would allow partial configuration updates & deletion policy (so that imported resources are not deleted on the remove).

**Comment by eahefnawy** [comment]
ah I think I get what you mean. It seems that the core of the issue is in the defaults, right? So what you're saying is that we should use defaults only when deploying new components? Otherwise use yaml config merged with existing config? Is that accurate?

**Comment by laardee** [comment]
Yep. Maybe if the inputs would have ARN specified then components would treat it as an imported component? And if I only want to refer to that function, I don't add any other parameters, and I can use component outputs in my other components.

```yaml
myFunction:
  component: "@serverless/aws-lambda"
  inputs:
    arn: 'arn:aws:lambda:us-east-1:123456789012:function:my-function'
```

These are just random ideas, but my main concern is that components should overwrite anything accidentally. If it only depends on the name of the component and then overwrites or modifies that resource it may cause issues. e.g. we have several hundreds of lambdas running in our env, it is impossible to know if some name is already used.

**Comment by eahefnawy** [comment]
@laardee I completely agree. It makes perfect sense. You're also striking a good balance between reusing existing resources and not messing things up.

What about removal though? if your component hooks up to an existing resource, when you run `components remove`, should that resource be removed?
